### PR TITLE
docs(ccp): how to use css custom properties

### PR DESCRIPTION
**Pull request by eduardoallegrini**

### CODE REVIEW

**Comment by eduardoallegrini** [comment]
Further work necessary: missing "how to use it in DXP?" section

File: css/how_to_use_css_custom_properties.md [hunk 1]
```diff
@@ -0,0 +1,154 @@
...
+}
+```
+
+[Advanced example](https://codepen.io/eduardoallegrini/pen/gOpdyKP)
+
+### By JS
+
+Another method to use CCP in your application is with JS
+
+The `element.style.getPropertyValue("-variable");` allow us to get the custom-property of the element, if the element doesn’t have a defined custom-property, the function will return an empty argument
```
Comment (p2kmgcl) [inline css/how_to_use_css_custom_properties.md]: Typo here: `--variable`

File: css/how_to_use_css_custom_properties.md [hunk 2]
```diff
@@ -0,0 +1,154 @@
+# How to use CCP (CSS Custom Properties)?
+
+## What are CCP?
+
+The CSS Custom Properties (CCP) are CSS variables that can be use and updated directly in the browser without having to be processed by a compiler like SCSS
```
Comment (wincent) [inline css/how_to_use_css_custom_properties.md]: can be use → can be used

Also, need a full stop at the end of this sentence.

File: css/how_to_use_css_custom_properties.md [hunk 3]
```diff
@@ -0,0 +1,154 @@
...
+
+## When to use CCP?
+
+The power of CCP is that it can be used in instances of the component.
+
+Let's say we have a general `.button` component and its default color is `red`
```
Comment (wincent) [inline css/how_to_use_css_custom_properties.md]: Needs trailing period.

File: css/how_to_use_css_custom_properties.md [hunk 4]
```diff
@@ -0,0 +1,154 @@
...
+
+We can use the same component and change the color by just declaring the variable in the component or its parent
```
Comment (wincent) [inline css/how_to_use_css_custom_properties.md]: Needs trailing period.

File: css/how_to_use_css_custom_properties.md [hunk 5]
```diff
@@ -0,0 +1,154 @@
...
+}
+```
+
+[Simple example](https://codepen.io/eduardoallegrini/pen/WNvgWMp)
+
+## How to use CCP?
+
+From the previous example we can main form of manipulate the CCP is through CSS itself.
+
+This method can be extended to the theme or section following the Cascade rule of CSS
```
Comment (wincent) [inline css/how_to_use_css_custom_properties.md]: Trailing period needed here.

Why is "Cascade" capitalized?

File: css/how_to_use_css_custom_properties.md [hunk 6]
```diff
@@ -0,0 +1,154 @@
...
+.component-3 {
+    --primary-color: green;
+}
+```
+
+[Simple example](https://codepen.io/eduardoallegrini/pen/WNvgWMp)
+
+## How to use CCP?
+
+From the previous example we can main form of manipulate the CCP is through CSS itself.
```
Comment (wincent) [inline css/how_to_use_css_custom_properties.md]: we can main form of manipulate → we can see the main form manipulating

File: css/how_to_use_css_custom_properties.md [hunk 7]
```diff
@@ -0,0 +1,154 @@
...
+		}
+	}
+}
+```
+
+[Advanced example](https://codepen.io/eduardoallegrini/pen/gOpdyKP)
+
+### By JS
+
+Another method to use CCP in your application is with JS
```
Comment (wincent) [inline css/how_to_use_css_custom_properties.md]: Trailing period.

File: css/how_to_use_css_custom_properties.md (hunk 1 above)
Comment (wincent) [reply css/how_to_use_css_custom_properties.md]: And we need a trailing period.

And drop the "The" at the beginning of the sentence.

File: css/how_to_use_css_custom_properties.md [hunk 8]
```diff
@@ -0,0 +1,154 @@
...
+
+The `element.style.getPropertyValue("-variable");` allow us to get the custom-property of the element, if the element doesn’t have a defined custom-property, the function will return an empty argument
+
+```JS
```
Comment (wincent) [inline css/how_to_use_css_custom_properties.md]: nit: all of these `JS` should be `js` as per [this list](https://github.com/github/linguist/blob/40992ba7f86889f80dfed3ba95e11e1082200bad/lib/linguist/languages.yml#L2501) (via [here](https://help.github.com/en/github/writing-on-github/creating-and-highlighting-code-blocks#fenced-code-blocks)).

File: css/how_to_use_css_custom_properties.md [hunk 9]
```diff
@@ -0,0 +1,154 @@
...
+// get variable from inline style
+element.style.getPropertyValue("--my-var");
+```
+
+The `getComputedStyle(element).getPropertyValue("--variable");` allow us to get the custom-property of the element, if the element doesn’t have a defined custom-property, the function will look for it at the parent elements or at the root of our markup
```
Comment (wincent) [inline css/how_to_use_css_custom_properties.md]: Drop initial "The", add trailing period.

You have a run-on sentence here that should be broken into two too:

> `getComputedStyle(element).getPropertyValue("--variable");` allow us to get the custom-property of the element. If the element doesn’t have a defined custom-property, the function will look for it at the parent elements or at the root of our markup.

File: css/how_to_use_css_custom_properties.md [hunk 10]
```diff
@@ -0,0 +1,154 @@
...
+
+```JS
+// get variable from wherever
+getComputedStyle(element).getPropertyValue("--my-var");
+```
+
+The `element.style.setProperty("--variable", "new-value");` allow us to set a new value for the custom-property of the element
```
Comment (wincent) [inline css/how_to_use_css_custom_properties.md]: Drop the "The", and add trailing period or colon.

File: css/how_to_use_css_custom_properties.md [hunk 11]
```diff
@@ -0,0 +1,154 @@
...
+```JS
+// set variable on inline style
+element.style.setProperty("--my-var", jsVar + 4);
+```
+
+[JS example](https://codepen.io/eduardoallegrini/pen/vYEGYZK)
+
+---
+
+Futher info can be found in the [Epic documentation](https://docs.google.com/document/d/1MKTfFLf6_zFhfHo0Ib2BCwAFXdxf3S0FOXO33zD1km8/edit#heading=h.fw7b9teo5ti6)
```
Comment (wincent) [inline css/how_to_use_css_custom_properties.md]: Futher → Further.

Also needs trailing comma.

**Comment by eduardoallegrini** [comment]
https://github.com/liferay/liferay-frontend-guidelines/pull/131

**Pull request closed without merging**